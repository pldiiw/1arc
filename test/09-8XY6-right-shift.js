'use strict';

let engine = require('engine').initialize();
const rightShift = require('instructions').rightShift;

engine.data[8] = 121;
engine.data[9] = 200;

const engine1 = rightShift(engine, 8, 8);
const engine2 = rightShift(engine1, 8, 9);

console.log(engine.data[8]);
console.log(engine.data[9]);
console.log(engine.data[0xF]);

console.log(engine1.data[8]);
console.log(engine1.data[9]);
console.log(engine1.data[0xF]);

console.log(engine2.data[8]);
console.log(engine2.data[9]);
console.log(engine2.data[0xF]);
