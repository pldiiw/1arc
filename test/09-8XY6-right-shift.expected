121
200
0
60
200
1
100
200
0
